public/igameevents.h

```diff
@@ -1,187 +1,225 @@
-
-	// create a new copy of this event, must be free later
-	virtual IGameEvent *DuplicateEvent( IGameEvent *event ) = 0;
-
-	// if an event was created but not fired for some reason, it has to bee freed, same UnserializeEvent
-	virtual void FreeEvent( IGameEvent *event ) = 0;
-
-	// write/read event to/from bitbuffer
-	virtual bool SerializeEvent( IGameEvent *event, CMsgSource1LegacyGameEvent *ev ) = 0;
-	virtual IGameEvent *UnserializeEvent( const CMsgSource1LegacyGameEvent &ev ) = 0; // create new KeyValues, must be deleted
-	
-	virtual int LookupEventId( const char *name ) = 0;
-	
-	virtual void PrintEventToString( IGameEvent *event, CUtlString &out ) = 0;
-
-	virtual bool HasEventDescriptor( const char *name ) = 0;
-};
-
-
-#endif // IGAMEEVENTS_H
+//========= Copyright � 1996-2005, Valve Corporation, All rights reserved. ============//
+//
+// Purpose: 
+//
+// $Workfile:     $
+// $Date:         $
+//
+//-----------------------------------------------------------------------------
+// $Log: $
+//
+// $NoKeywords: $
+//=============================================================================//
+#if !defined( IGAMEEVENTS_H )
+#define IGAMEEVENTS_H
+#ifdef _WIN32
+#pragma once
+#endif
+
+#include "interfaces/interfaces.h"
+#include "tier1/bitbuf.h"
+#include "tier1/generichash.h"
+#include "tier1/utlstring.h"
+
+class CMsgSource1LegacyGameEvent;
+class CPlayerSlot;
+class CBasePlayer;
+class CEntityIndex;
+class CEntityHandle;
+class CBaseEntity;
+//-----------------------------------------------------------------------------
+// Purpose: Engine interface into global game event management
+//-----------------------------------------------------------------------------
+
+/* 
+
+The GameEventManager keeps track and fires of all global game events. Game events 
+are fired by game.dll for events like player death or team wins. Each event has a
+unique name and comes with a KeyValue structure providing informations about this
+event. Some events are generated also by the engine.
+
+Events are networked to connected clients and invoked there to. Therefore you
+have to specify all data fields and there data types in an public resource
+file which is parsed by server and broadcasted to it's clients. A typical game 
+event is defined like this:
+
+	"game_start"				// a new game starts
+	{
+		"roundslimit"	"long"		// max round
+		"timelimit"		"long"		// time limit
+		"fraglimit"		"long"		// frag limit
+		"objective"		"string"	// round objective
+	}
+
+All events must have unique names (case sensitive) and may have a list
+of data fields. each data field must specify a data type, so the engine
+knows how to serialize/unserialize that event for network transmission.
+Valid data types are string, float, long, short, byte & bool. If a 
+data field should not be broadcasted to clients, use the type "local".
+*/
+
+struct GameEventKeySymbol_t
+{
+	inline GameEventKeySymbol_t(const char* keyName)
+	{
+		m_nHashCode = 0;
+		m_pszKeyName = keyName;
```

Comment:
Also these could be moved to an initializer list instead

Reply:
You haven't moved them to initializer list, but removed, that's not really what was asked?